### PR TITLE
Date formatter performance

### PR DESCRIPTION
**Pull request by rahimkhalid**
## Checklist

- [ ] New Extension
- [ ] New Test
- [ ] Changed more than one extension, but all changes are related
- [x] Trivial change (doesn't require changelog)

### CODE REVIEW

**Comment by rahimkhalid** [comment]
@Khalian @goktugyil can you kindly add reviewers for this PR.

File: Sources/DateExtensions.swift [hunk 1]
```diff
@@ -8,27 +8,38 @@
 
 import Foundation
 
+/// EZSE: This Dict help to cache already created formatter to use them in future, helps in performace improvement.
+fileprivate var dateFormatterDict: [String:DateFormatter] = [:]
```
Comment (Khalian) [inline Sources/DateExtensions.swift]: This entity needs synchronization.

File: EZSwiftExtensionsTests/DateTests.swift [hunk 2]
```diff
@@ -73,7 +73,7 @@ class DateTests: XCTestCase {
 
     func testDateToString() {
         let date = Date(timeIntervalSince1970: 0)
-
+        let format = "yyyy-MM-dd"
```
Comment (Khalian) [inline EZSwiftExtensionsTests/DateTests.swift]: Could you add tests here to verify the size of the dictionary increasing and decreasing as needed. We want to validate at unit level that the right objects were indeed cached.

File: Sources/DateExtensions.swift [hunk 3]
```diff
@@ -213,41 +242,51 @@ extension Date {
...
     }
-
+    
     /// EZSE : Get the weekday from the date
     public var weekday: String {
-        let dateFormatter = DateFormatter()
-        dateFormatter.dateFormat = "EEEE"
-        return dateFormatter.string(from: self)
+        let format = "EEEE"
+        if let dateFormatter = dateFormatterDict[format]{
```
Comment (Khalian) [inline Sources/DateExtensions.swift]: This guarded check is used in a whole bunch of places. Can you please make a private method to put the logic for getting the date formattter in one place?

File: Sources/DateExtensions.swift (hunk 1 above)
Comment (rahimkhalid) [reply Sources/DateExtensions.swift]: I have updated the pull request kindly review @Khalian

File: EZSwiftExtensionsTests/DateTests.swift (hunk 2 above)
Comment (rahimkhalid) [reply EZSwiftExtensionsTests/DateTests.swift]: I have updated the pull request kindly review @Khalian

File: Sources/DateExtensions.swift (hunk 3 above)
Comment (rahimkhalid) [reply Sources/DateExtensions.swift]: I have updated the pull request kindly review @Khalian 
I have created a method for getting formatters from dict.

File: Sources/DateExtensions.swift (hunk 1 above)
Comment (rahimkhalid) [reply Sources/DateExtensions.swift]: I think this is now synchronized, and is ready to be merged.

File: EZSwiftExtensionsTests/DateTests.swift (hunk 2 above)
Comment (rahimkhalid) [reply EZSwiftExtensionsTests/DateTests.swift]: Unit test added to test the `dict` functionality.

**Comment by rahimkhalid** [comment]
@Khalian I think this pull request is now ready to be merged.

**Pull request merged**